src/core/i18n/utils.js

```diff
@@ -80,3 +80,17 @@ export function isRtlLang(lang) {
 export function getDirection(lang) {
   return isRtlLang(lang) ? 'rtl' : 'ltr';
 }
+
+export function getLangFromRouter(renderProps) {
+  let lang = defaultLang;
```

Comment:
I think this would be clearer with multiple returns rather than the var.

```js
export function getLangFromRouter(renderProps) {
 if (renderProps && renderProps.params) {
    if (renderProps.params.lang) {
      return getLanguage(renderProps.params.lang);
    } else if (renderProps.params.query && renderProps.params.query.lang) {
      return getLanguage(renderProps.params.query.lang);
    }
  }
  return defaultLang;
}